plugins/BEdita/Core/src/Model/Behavior/CustomPropertiesBehavior.php

```diff
@@ -112,13 +112,16 @@ public function getDefaultValues()
      * @param \Cake\ORM\Query $query Query object instance.
      * @return void
      */
-    public function beforeFind(Event $event, Query $query)
+    public function beforeFind(Event $event, Query $query): Query
     {
-        $query->formatResults(function (CollectionInterface $results) {
-            return $results->map(function ($row) {
-                return $this->promoteProperties($row);
-            });
-        });
+        return $query->formatResults(
```

Comment:
Do we need to `return` the Query object? New formatter is added to `$query` object in use.

Also `@return void` annotation conflicts with `Query` return type.

Reply:
No, it is not strictly needed since it is modified in place. However, I personally prefer to `return` it because it better complies with a functional style — i.e.: it will continue to work even if the `formatResult()` method will start returning a _copy_ of the Query object at some point — which Cake has started adopting in many cases (e.g. Request and Response objects, etc.).

You're right about the annotation. I'm fixing it.